File: java/src/tk/csparpa/gofdp/builder/Product.java
package tk.csparpa.gofdp.builder;

/**
 * 
 * This class represents the complex object to be built
 *
 */
public class Product {
	
	private String partA;
	private int partB;
	
	public void addPartA(String part){
		System.out.println("Adding: partA to Product instance");
		this.partA = part;
	}
	
	public void addPartB(int part){
		System.out.println("Adding: partB to Product instance");
		this.partB = part;
	}
}

File: python/builder/product.py
"""""
This class represents the complex object to be built
"""
class Product():
    
    _part_A = None
    _part_B = None
    
    def add_part_A(self, part):
        print("Adding: partA to Product instance")
        self._part_A = part
    
    def add_part_B(self, part):
        print("Adding: partB to Product instance")
        self._part_B = part
